Mathlib/Algebra/Group/Pointwise/Set/Basic.lean

```diff
@@ -240,6 +243,18 @@ theorem inv_range {ι : Sort*} {f : ι → α} : (range f)⁻¹ = range fun i =>
   rw [← image_inv_eq_inv]
   exact (range_comp ..).symm
 
+@[to_additive]
+theorem image_inv_of_apply_inv_eq {f g : α → β} (H : ∀ x ∈ s, f (x⁻¹) = g x) :
+    f '' (s⁻¹) = g '' s := by
+  ext
+  rw [mem_image, ← (Equiv.inv _).exists_congr_right]
+  aesop
```

Comment:
I'm not convinced about this one. You can prove it inline very quickly:
```suggestion
@[to_additive]
theorem image_inv_of_apply_inv_eq {f g : α → β} (H : ∀ x ∈ s, f (x⁻¹) = g x) :
    f '' (s⁻¹) = g '' s := by
  rw [Set.image_inv_eq_inv, Set.image_image]; exact Set.image_congr H
```

Reply:
The golf is very welcome, but I still think the theorem is a useful shortcut. We currently use it 5 times on the CGT repo!